### PR TITLE
Copy IAM authentication state from status

### PR DESCRIPTION
**Pull request by arjunrn**
Issue #, if available: #1937

Description of changes: Like the parameter groups the IAM authentication flag has different names in the input and output to the API. A hook is need to correctly set the value.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_db_cluster.py [hunk 1]
```diff
@@ -108,6 +108,53 @@ def aurora_mysql_cluster(k8s_secret):
...
+
+@pytest.fixture
+def aurora_postgres_cluster(k8s_secret):
+    db_cluster_id = random_suffix_name("my-aurora-postgres", 20)
+    secret = k8s_secret(
+        MUP_NS,
+        MUP_SEC_NAME,
+        MUP_SEC_KEY,
+        MUP_SEC_VAL,
+    )
```
Comment (a-hilaly) [inline test/e2e/tests/test_db_cluster.py:120]: Thanks @arjunrn ! We should use a different name here to avoid conflict with other created secrets

Comment (arjunrn) [reply test/e2e/tests/test_db_cluster.py]: I've fixed the naming. PTAL.

**Review by a-hilaly — APPROVED** [review]
Thanks @arjunrn !
/lgtm

**Pull request merged**